compiler/bcq-tools/generate_bcq_output_arrays

```diff
@@ -87,7 +87,154 @@ def find_bcq_version(flags):
+    # it is used for applying BCQ to constant node named PREFIX_1.
+    # Collected prefixes will be used for connecting
+    # bcqinfo nodes and prefix node user operations.
+    prefix_set = set()
+    has_dequant_weight = False
+    for op in ops:
+        if op.type == "Const" and "/bcqinfo_" in op.outputs[0].name:
+            # Metadata do not have prefix
+            if "one_compiler/bcqinfo_one_metadata" in op.outputs[0].name:
+                continue
+
+            prefix_index = op.outputs[0].name.index("/bcqinfo_")
+            prefix = op.outputs[0].name[:prefix_index]
+            prefix_set.add(prefix)
+
+            infoname = op.outputs[0].name[prefix_index + 1:-2]
+            if infoname == "bcqinfo_dequant_weight":
+                has_dequant_weight = True
+
+    # Ideal situation is that the users of BCQ applicable constant nodes
+    # are BCQ applicable operations such as MatMul, GatherV2, etc.
+    # However, operations which do not change original values such as
+    # Ideneity or Transpose can exist between them. In view of TensorFlow Lite,
+    # real users of BCQ applicable constant nodes must be found first.
+    # This work is done by BFS search with queue.
+
+    prefix_node_dict = {}  # key : prefix / value : list of candidates
+    matmul_node_prefix_dict = {}  # key : Name of MatMul node / value : prefix
+
+    queue_prefix = list(prefix_set)
+    queue_nodename = [queue_prefix[idx] + ":0" for idx in range(len(queue_prefix))]
+
+    while len(queue_prefix) > 0:
+        prefix = queue_prefix.pop(0)
+        nodename = queue_nodename.pop(0)
+        if prefix not in prefix_node_dict.keys():
+            prefix_node_dict[prefix] = []
+
+        for op in ops:
+            if op.type == "MatMul" and (op.inputs[0].name == nodename
+                                        or op.inputs[1].name == nodename):
+                prefix_node_dict[prefix].append(op.outputs[0].name[:-2])
+                matmul_node_prefix_dict[op.outputs[0].name[:-2]] = prefix
+            elif op.type == "Einsum" and (op.inputs[0].name == nodename
+                                          or op.inputs[1].name == nodename):
+                prefix_node_dict[prefix].append(op.outputs[0].name[:-2])
+            elif op.type == "GatherV2" and op.inputs[0].name == nodename:
+                prefix_node_dict[prefix].append(op.outputs[0].name[:-2])
+            elif len(op.outputs) == 1:
+                for i in range(len(op.inputs)):
+                    if op.inputs[i].name == nodename:
+                        queue_prefix.append(prefix)
+                        queue_nodename.append(op.outputs[0].name)
+                        break
+
+    # When TensorFlow model is converted to TensorFlow Lite model,
+    # more than one operation can be fused as one.
+    # For example, MatMul + BiasAdd + ReLU in TensorFlow can be fused as
+    # one FullyConnected in TensorFlow Lite.
+    # It means that even real users of BCQ applicable constant nodes
```

Comment:
Is `users` here refer to human users? If not, what is the `users` here? maybe `using node(s)` ?

Reply:
> maybe using node(s)

This is correct, I will fix it :)